### PR TITLE
Fixed f8_gemm NaN

### PR DESCRIPTION
**Pull request by zjing14**
- workaround nan problem by changing output to fp16

### CODE REVIEW

**Review by illsilin — DISMISSED** [review]
Verified that the tests are now working:

        Start   9: example_gemm_xdl_fp8
  9/193 Test   #9: example_gemm_xdl_fp8 ......................................................   Passed   13.75 sec
        Start  10: example_gemm_xdl_fp8_bf8
 10/193 Test  #10: example_gemm_xdl_fp8_bf8 ..................................................   Passed   13.65 sec
        Start  11: example_gemm_xdl_fp16_fp8
 11/193 Test  #11: example_gemm_xdl_fp16_fp8 .................................................   Passed   20.57 sec

File: include/ck/utility/type_convert.hpp [hunk 1]
```diff
@@ -393,7 +393,7 @@ inline __host__ __device__ bf8_t f8_convert_sr<bf8_t, half_t>(half_t x)
 #if defined(__gfx940__) || defined(__gfx941__) || defined(__gfx942__)
     // convert to float and use native converion
     return f8_convert_sr<f8_t>(type_convert<float>(x));
```
Comment (geyyer) [inline include/ck/utility/type_convert.hpp:397]: @zjing14 could you also fix f8_t -> bf8_t here?

Comment (zjing14) [reply include/ck/utility/type_convert.hpp:397]: f8->bf8_t?

Comment (geyyer) [reply include/ck/utility/type_convert.hpp:397]: Yeah, it is probably a typo. The method converts half_t to bf8_t, nothing to do with f8_t.

**Review by geyyer — APPROVED** [review]
LGTM!

**Pull request merged**